11-Backbonejs/Views-of-Collections.md

```diff
@@ -0,0 +1,208 @@
+
+![TaskListView](images/Ada-TaskList-JS.png)
+
+## Learning Goals
+-  Create a view to manage the entire TaskList collection
+-  Create event listeners to add new models to the collection
+-  Create event listeners for custom Backbone Events
+-  Create a `render` method which creates and renders a group of smaller views.  
+
+## Getting Started
+
+This view starts like `TaskView` with a file named `/src/views/task_list_view.js`.  We can move our renderList code into the view's render method and make a few adjustments.
+
+```javascript
+import Backbone from 'backbone';
+import _ from 'underscore';
+import $ from 'jquery';
+import TaskView from '../views/task_view.js';
+
+
+var TaskListView = Backbone.View.extend({
+  initialize: function(params) {
+    this.template = params.template;
+  },
+  render: function() {
+    // Clear the unordered list
+    this.$('.todo-items').empty();
+    // Iterate through the list rendering each Task
+    var that = this;
+    this.model.each(function(task) {
+      // Create a new TaskView with the model & template
+      var taskView = new TaskView({
+        model: task,
+        template: that.template,
+        tagName: 'li'
+      });
+      // Then render the TaskView
+      // And append the resulting HTML to the DOM.
+      that.$('.todo-items').append(taskView.render().$el);
+    });
+    return this;
+  }
+});
+
+export default TaskListView;
+```
+Just like the `TaskView` in `initialize` we store the template in an attribute of the view.  The view's render method looks like the render method we wrote earlier with a few differences particularly with regard to `this` and `that`.
+
+### A Note about `this` and `that`.  
+
+If you'll notice we save our current context `this` into a variable named `that` in the render method.  We do this because the `.each` method's callback method is called within the context of the collection.  
+
+We finish the `render` method by appending each TaskView to the list inside the DOM and when the loop is finished we `return this;` by convention.  It's a fairly common pattern developers use in JavaScript to deal with confusing `this` issues. 
+
+
+### About `this.$`
+
+In the `render` method we clear the list using `this.$`.  The method `this.$` performs a jQuery selection for HTML inside `el`.  Using `this.$` we can ensure that our view does not interact with any code outside of it.  You should never use jQuery directly within a view.  Instead you should use `this.$` just as you would jQuery to avoid impacting the DOM outside your view.  
+
+
+### Updating `app.js`
+
+We will adjust our `$(document).ready` handler to use the `TaskView`.
+
+
+```JavaScript
+// app.js
+// imports etc...
+
+import TaskListView from './views/task_list_view';
+
+// ...
+
+$(document).ready(function() {
+  var taskData = [{
```

Comment:
As before, I would think of this as something that should be defined outside of `$(document).ready`, probably up at the top of the file. Just an opinion though.